src/Dhall/Binary.hs

```diff
@@ -262,16 +268,10 @@ encode_1_1 (Some t₀) =
     TList [ TInt 5, TNull, t₁ ]
   where
     t₁ = encode_1_1 t₀
-encode_1_1 (Merge t₀ u₀ Nothing) =
-    TList [ TInt 6, t₁, u₁ ]
-  where
-    t₁ = encode_1_1 t₀
-    u₁ = encode_1_1 u₀
-encode_1_1 (Merge t₀ u₀ (Just _T₀)) =
-    TList [ TInt 6, t₁, u₁, _T₁ ]
+encode_1_1 (Merge _T₀ t₀) =
+    TList [ TInt 6, t₁, _T₁ ]
```

Comment:
But the 2-element encoding conflicts – it used to mean no type annotation, but now it means partially-applied.

I couldn’t quickly come up with anything that wasn’t incompatible with the old version.